internal/pkg/client/git/client.go

```diff
@@ -75,3 +81,41 @@ func (client *Client) Parse(rawURL string) *url.ParsedURL {
 		Fragment: results["fragment"],
 	}
 }
+
+func cloneRepo(parsedRepoURL *url.ParsedURL, auth *url.BasicAuth,
+	opts *options.Options,
+) (*git.Repository, string, error) {
+	// Create temp directory to clone into.
+	tempDir, err := os.MkdirTemp(os.TempDir(), strings.ReplaceAll(parsedRepoURL.Path, "/", "-"))
```

Comment:
It appears we do not cleanup/delete the temp directory after we are done with it. Is that intentional?

https://pkg.go.dev/os#MkdirTemp
>It is the caller's responsibility to remove the directory when it is no longer needed.

Reply:
Somewhere after line 92, we would want to add `defer os.RemoveAll(dir) // clean up`

Reply:
So we are calling it, but from the functions that call this function. I can have this clone repo function return the defer function if that sounds better. 

in the git client code, in push.gp (push): Line 58
also in the git client fetch (fetch): Line 45

Reply:
It might make more sense to add it to this function to create a more self contained code unless it doesn't make sense.

Reply:
I removed the mktempdir from clone repo and added it into fetch and push, and updated clone repo to get the tempdir path passed in, that way the creation and removal are all in the same function. Easier to read/follow